src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/pipeline/JiraSendBuildInfoStep.java

```diff
@@ -100,9 +106,16 @@ public JiraSendBuildInfoStepExecution(
         @Override
         protected JiraSendInfoResponse run() throws Exception {
             final TaskListener taskListener = getContext().get(TaskListener.class);
-            final WorkflowRun build = getContext().get(WorkflowRun.class);
+            final WorkflowRun workflowRun = getContext().get(WorkflowRun.class);
+            final Optional<String> site = pickDefaultSite(step.getSite());
+
+            if (!site.isPresent()) {
+                workflowRun.setResult(Result.FAILURE);
+                getContext().setResult(Result.FAILURE);
+                throw new DefaultSiteLookupFailureException();
```

Comment:
It seems like throwing exception is the only way to fail the build. I tried setting results (see two lines above) to failure but that didn't stop the build.

Reply:
Our plugin is a build post-action, it can't change the build status (i.e. setResult) and in my opinion it shouldn't fail a build. I think it should only log a warning or error to the console - i.e. the same what we do now when no Jira sites are defined.